op-chain-ops/cmd/celo-migrate/main.go

```
@@ -515,116 +517,47 @@ func runDBCheck(opts dbCheckOptions) (err error) {
 		err = errors.Join(err, nonAncientDB.Close())
 	}()
 
-	lastAncient, err := loadLastAncient(ancientDB)
+	ancients, err := ancientDB.Ancients()
 	if err != nil {
-		return fmt.Errorf("failed to load last ancient block: %w", err)
+		return fmt.Errorf("failed to call Ancients: %w", err)
 	}
-	lastAncientNumber := lastAncient.Header().Number.Uint64()
-	lastBlockNumber := *rawdb.ReadHeaderNumber(nonAncientDB, rawdb.ReadHeadHeaderHash(nonAncientDB))
-
-	var errResult *multierror.Error
-
-	// First, check continuity between ancients and non-ancients.
-	// Gaps in data will often halt the freezing process, so attempting to load the first non-ancient block
-	// will most likely fail if there is a gap.
-	firstNonAncientRange, err := loadNonAncientRange(nonAncientDB, lastAncientNumber+1, 1)
+	// Check the range covering the ancient to non-ancient transition first, since this is where we previously encountered continuity issues.
+	err = checkRange(ancientDB, nonAncientDB, ancients-1, ancients)
 	if err != nil {
-		if opts.failFast {
-			return fmt.Errorf("failed to load first non-ancient block: %w", err)
-		}
-		// We don't need to add the error to errResult here because it will be added below when we call checkContinuity on non-ancients
-	} else {
-		if _, err := firstNonAncientRange.CheckContinuity(lastAncient, 1); err != nil {
-			err = fmt.Errorf("failed continuity check between ancients and non-ancients: %w", err)
-			if opts.failFast {
-				return err
-			}
-			errResult = multierror.Append(errResult, err)
-		}
+		return fmt.Errorf("failed to check range covering ancient to non-ancient transition (blocks %d-%d): %w", ancients-1, ancients+1, err)
```

For the fail fast logic top work we also need to add it here